TCAT/Controllers/AllStopsTableViewController.swift

```diff
@@ -117,40 +112,115 @@ class AllStopsTableViewController: UITableViewController {
             // Adding "#" to keys for bus stops that start with a number
             sortedKeys.append("#")
         }
+    }
 
-        return sortedKeys
+    @objc private func backAction() {
+        navigationController?.popViewController(animated: true)
     }
 
-    func setUpTableOnRetry() {
-        // Retry getting data from user defaults
-        self.allStops = SearchTableViewManager.shared.getAllStops()
-        // Set up table information
-        self.sectionIndexes = self.sectionIndexesForBusStop()
-        self.sortedKeys = self.sortedKeysForBusStops()
+    private func setUpLoadingIndicator() {
+        loadingIndicator = LoadingIndicator()
+        if let loadingIndicator = loadingIndicator {
+            view.addSubview(loadingIndicator)
+            loadingIndicator.snp.makeConstraints { make in
+                make.center.equalToSuperview()
+                make.width.height.equalTo(40)
+            }
+        }
+    }
+
+    private func getAllStops() -> Future<Response<[Place]>> {
+        return networking(Endpoint.getAllStops()).decode()
+    }
 
-        self.tableView.reloadData()
+    /* Get all bus stops and store in userDefaults */
+    private func refreshAllStops() {
+        setUpLoadingIndicator()
+        if let allBusStops = userDefaults.value(forKey: Constants.UserDefaults.allBusStops) as? Data,
+            var busStopArray = try? decoder.decode([Place].self, from: allBusStops) {
+            // Check if empty so that an empty array isn't returned
+            if !busStopArray.isEmpty {
+                // TODO: Move to backend
+                // Creating "fake" bus stop to remove Google Places central Collegetown location choice
+                let collegetownStop = Place(name: "Collegetown", latitude: 42.442558, longitude: -76.485336)
+                busStopArray.append(collegetownStop)
+                allStops = busStopArray
+            }
+            loadingIndicator?.removeFromSuperview()
+            loadingIndicator = nil
+            createSectionIndexesForBusStop()
+            tableView.reloadData()
+        } else {
```

Comment:
you could add a `return` after line 152 and then remove this `else` block so that the code isn't indented